samples/nfc/tnep_tag/src/main.c

```diff
@@ -0,0 +1,240 @@
+static const u8_t en_code[] = {'e', 'n'};
+
+static const char svc_one_msg[] = "Service pi = 3.14159265358979323846";
+static const char svc_two_msg[] = "Service e  = 2.71828182845904523536";
+
+static const u8_t svc_one_uri[] = "svc:pi";
+static const u8_t svc_two_uri[] = "svc:e";
+
+static u8_t tag_buffer[NDEF_TNEP_MSG_SIZE];
+static u8_t tag_buffer2[NDEF_TNEP_MSG_SIZE];
+
+NFC_NDEF_MSG_DEF(ndef_msg, 16);
+
+NFC_NDEF_TEXT_RECORD_DESC_DEF(nfc_text, UTF_8, en_code, sizeof(en_code),
+			      msg, sizeof(msg));
+NFC_NDEF_TEXT_RECORD_DESC_DEF(svc_one_rec, UTF_8, en_code, sizeof(en_code),
+			      svc_one_msg, sizeof(svc_one_msg));
+NFC_NDEF_TEXT_RECORD_DESC_DEF(svc_two_rec, UTF_8, en_code, sizeof(en_code),
+			      svc_two_msg, sizeof(svc_two_msg));
+
+
+static struct k_poll_event events[NFC_TNEP_EVENTS_NUMBER];
+
+static void tnep_svc_one_selected(void)
+{
+	int err;
+
+	printk("Service one selected\n");
+
+	err = nfc_tnep_tag_tx_msg_app_data(&NFC_NDEF_TEXT_RECORD_DESC(svc_one_rec),
+					   1, NFC_TNEP_STATUS_SUCCESS);
+	if (err) {
+		printk("Service app data set err: %d\n", err);
+	}
+
+	dk_set_led_on(LED_SVC_ONE);
+}
+
+static void tnep_svc_one_deselected(void)
+{
+	printk("Service one deselected\n");
+
+	dk_set_led_off(LED_SVC_ONE);
+}
+static void tnep_svc_one_msg_received(const u8_t *data, size_t len)
+{
+	int err;
+
+	printk("New service data received. Finish service data exchange\n");
+
+	err = nfc_tnep_tag_tx_msg_no_app_data();
+	if (err) {
+		printk("TNEP Service data finish err: %d\n", err);
+	}
+}
+
+static void tnep_svc_error(int err_code)
+{
+	printk("Service data exchange error: %d\n", err_code);
+}
+
+static void tnep_svc_two_selected(void)
+{
+	printk("Service two selected\n");
+
+	dk_set_led_on(LED_SVC_TWO);
+}
+
+static void tnep_svc_two_deselected(void)
+{
+	printk("Service two deselected\n");
+
+	dk_set_led_off(LED_SVC_TWO);
+}
+static void tnep_svc_two_msg_received(const u8_t *data, size_t len)
+{
+	int err;
+
+	printk("New service data received. Finish service data exchange\n");
+
+	err = nfc_tnep_tag_tx_msg_no_app_data();
+	if (err) {
+		printk("TNEP Service data finish err: %d\n", err);
+	}
+}
+
+NFC_TNEP_TAG_SERVICE_DEF(svc_one, svc_one_uri, (ARRAY_SIZE(svc_one_uri) - 1),
+		     NFC_TNEP_COMM_MODE_SINGLE_RESPONSE, 20, 4, 1024,
+		     tnep_svc_one_selected, tnep_svc_one_deselected,
+		     tnep_svc_one_msg_received, tnep_svc_error);
+
+NFC_TNEP_TAG_SERVICE_DEF(svc_two, svc_two_uri, (ARRAY_SIZE(svc_two_uri) - 1),
+		     NFC_TNEP_COMM_MODE_SINGLE_RESPONSE, 63, 4, 1024,
+		     tnep_svc_two_selected, tnep_svc_two_deselected,
+		     tnep_svc_two_msg_received, tnep_svc_error);
+
+static struct nfc_tnep_tag_service training_services[] = {
+	NFC_TNEP_TAG_SERVICE(svc_one),
+	NFC_TNEP_TAG_SERVICE(svc_two),
+};
+
+static void nfc_callback(void *context, enum nfc_t4t_event event,
+			 const u8_t *data, size_t data_length, u32_t flags)
+{
+	switch (event) {
+	case NFC_T4T_EVENT_NDEF_READ:
+		break;
```

Comment:
What is the reason of leaving non-processed cases? If this is for example - just add an comment what may be processed here.

Reply:
Removed. They are no needed for any reason